lib/benchee/formatters/csv.ex

```diff
@@ -86,32 +77,33 @@ defmodule Benchee.Formatters.CSV do
   defp get_benchmarks_raw(scenarios) do
     scenarios
     |> Enum.map(fn scenario -> scenario.run_times end)
-    |> Util.zip_all
+    |> Util.zip_all()
     |> Raw.add_headers(scenarios)
     |> CSV.encode()
   end
 
-  @default_filename "benchmark_output.csv"
-  defp get_filename(%Configuration{formatter_options: %{csv: %{file: filename}}}), do: filename
-  defp get_filename(_configuration), do: @default_filename
-
   @doc """
-  Uses the return value of `Benchee.Formatters.CSV.format/1` to write the
+  Uses the return value of `Benchee.Formatters.CSV.format/2` to write the
   statistics output to a CSV file, defined in the initial
-  configuration under `[formatter_options: [csv: [file: \"my.csv\"]]`.
-  If file is not defined then output is going to be placed in
-  "benchmark_output.csv". All raw measurements are placed in raw_benchmark_output.csv.
+  configuration. The raw measurements are placed in a file with "raw_" prepended
+  onto the file name given in the initial configuration.
+
+  If no file name is given in the configuration, "benchmark_output.csv" is used
+  as a default.
   """
-  @spec write([{Enumerable.t, String.t}]) :: :ok
-  def write(benchmarks) do
-    Enum.each(benchmarks, fn({content, filename}) -> write(content, filename) end)
+  @spec write({Enumerable.t(), Enumerable.t()}, map | nil) :: :ok
+  def write(data, nil), do: write(data, %{})
+  def write({statistics, raw_measurements}, options) do
```

Comment:
it's not obvious to me why we'd need the nil case?

Reply:
ah makes total sense, I think it should more likely be called with an empty map instead of nil thoug... we could revisit that in benchee.

It's the case when no options are specified, but I think an empty hash represents that better, what do you think?

Reply:
I do agree. I _thought_ that’s what we were doing already, but I’m gonna have to revisit that.

Reply:
I thought so as well so this is super odd :man_shrugging: 

Let's revisit in benchee and let this pass for now